packages/opds-web-client/src/components/DownloadButton.tsx

```diff
@@ -1,89 +1,41 @@
 import * as React from "react";
-import download from "./download";
-import { useActions } from "./context/ActionsContext";
-import { typeMap, generateFilename } from "../utils/file";
+import useDownloadButton from "../hooks/useDownloadButton";
+import { MediaLink, FulfillmentLink } from "./../interfaces";
 
-export interface DownloadButtonProps extends React.HTMLProps<{}> {
-  url: string;
-  mimeType: string;
-  isPlainLink?: boolean;
-  title?: string;
-  indirectType?: string;
-}
+export type DownloadButtonProps = React.ComponentProps<"button"> &
```

Comment:
Why the switch from `interface` to `type`?

Reply:
Was easier for me to union with multiple types. I don't think inheriting multiple interfaces works the same as this. I find types map better to modern JS than interfaces, since they're not using inheritance when it isn't necessary.